vignettes/customization.Rmd

```diff
@@ -0,0 +1,266 @@
+
+### Tweaking other templates (advanced users only)
+
+The main difference compared to using empty templates meant for customization is that you will need to learn a bit about whisker, the R implementation of [Mustache](https://mustache.github.io/mustache.5.html); and that you might need to update your templates with upstream changes in the future.
+
+* To find what template to edit, study the contents of pkgdown's templates folder (`inst/templates/BS4` for Bootstrap 4, `inst/templates/BS3` for Bootstrap 3). The skeleton of pages is provided in "layout.html" so start there.
+* Create a folder `inst/pkgdown/templates` (or use another method to override templates).
+* Copy the template(s) you need to tweak into that folder, edit and save them.
+* Re-build your website, inspect it (and its source, if needed).
+
+### General caveat
+
+A limitation to tweaking templates is that, for anything not hard-coded, your templates can only use data that pkgdown passes to templates.
+Make sure you read the documentation of `render_page()` as well as its source code.
+
+## Sharing pkgdown styles: template packages
+
+To share a pkgdown style across several packages, the best workflow is to create... a package!
+
+It might contain any of the following, under `inst/pkgdown`
+
+* a configuration file in `inst/pkgdown/_pkgdown.yml`;
+* templates in `inst/pkgdown/templates/BS4`;
+* assets in `inst/pkgdown/assets/BS4`.
+
+Once you have created your template package `theverybest`, to build a pkgdown website with it, what's needed is
+
+* for `theverybest` to be installed. In `DESCRIPTION` the packages using it could have these lines (in particular if your continuous integration workflow uses pak or remotes).
+
+```
+Config/Needs/website:
+    theverybest
+```
+
+* for the pkgdown configuration file to feature these lines:
+
+```yaml
+template:
+  package: theverybest
+```
+
+### Configuration file in a template package
+
+Allow template packages to provide a default configuration `_pkgdown.yml`, 
+stored in `inst/pkgdown/_pkgdown.yml`. This can be used to set (e.g.) author 
+definitions, Bootstrap version and variables, the sidebar, footer, navbar, etc.
+
+Configuration parameters supplied by a template package have the lowest priority
+They will be overridden by a local `_pkgdown.yml` which is in turn overridden by
+`override` parameter of `build_site()`. 
+
+### Examples
+
+Note that the pkgdown website of pkgdown itself is customized as little as possible.
+
+Here is a list of examples of template packages that you can use as inspiration but that you cannot use directly as they are aimed at a specific subset of packages.
+
+* [tidytemplate](https://tidytemplate.tidyverse.org/) used for tidyverse and tidymodels packages;
+* [quillt](https://pkgs.rstudio.com/quillt) used for R Markdown packages;
+* [rotemplate](https://github.com/ropensci-org/rotemplate) used for rOpenSci packages.
+
+## How to craft and preview style tweaks
```

Comment:
I wonder whether there should be something about adding a pkgdown preview GHA workflow for the repo of template packages.

Reply:
I think you could just link to a couple of example pkgdown actions so that people can see what we do.